amazonka/src/Network/AWS/Auth.hs

```diff
@@ -98,10 +99,14 @@ envSecretKey = "AWS_SECRET_ACCESS_KEY"
 envSessionToken :: Text -- ^ AWS_SESSION_TOKEN
 envSessionToken = "AWS_SESSION_TOKEN"
 
--- | Credentials profile environment variable.
+-- | Default credentials profile environment variable.
 envProfile :: Text -- ^ AWS_PROFILE
 envProfile = "AWS_PROFILE"
 
+-- | Default region environment variable
+envRegion :: Text -- ^ AWS_REGION
+envRegion = "AWS_REGION"
```

Comment:
The documentation you linked is for the Python CLI.

All of the offical SDKs use `AWS_REGION` as follows:

* https://github.com/aws/aws-sdk-js/blob/5b3a8a69f876f8b4f52e422e0a076cb2e3855c48/lib/aws.js#L56
* https://github.com/aws/aws-sdk-java/blob/ebd165684ae3857c71796512824eb850d22d8830/aws-java-sdk-core/src/main/java/com/amazonaws/SDKGlobalConfiguration.java#L191-L194

And the Go SDK has a comment about `AWS_REGION` vs `AWS_DEFAULT_REGION`:

* https://github.com/aws/aws-sdk-go/blob/f34b74c96bfd27df35643adeb14d8431ca047df5/aws/session/env_config.go#L32-L41

> AWS_DEFAULT_REGION is only read if AWS_SDK_LOAD_CONFIG is also set,  and AWS_REGION is not also set.